### ISSUE TITLE
Is the calculation of loss influenced by I_{HQ} during the training of LAControlNet?

### ISSUE THREAD

**Issue by SongYxing** [XPixelGroup/DiffBIR#52]
I_{HQ} only works during the SwinIR training?  I couldn't locate any involvement of I_{HQ} in the LAControlNet training code. Whether "|I_{HQ} - I_{reg}|" is computed as part of the LAControlNet training process? LAControlNet only need to I_{reg} as input? If I intend to directly train LAControlNet using I_{HQ} and I_{reg}, will {I_HQ} be incorporated in the process?

**Comment by 0x3f3f3f3fun** [comment]
Hello! $I_{HQ}$ will be used in both stage1 (SwinIR) and stage2 (LAControlNet) training. LAControlNet requires two inputs: the noisy latent $z_t$ and the condition latent $\epsilon(I_{reg})$, $\epsilon$ denotes the pretrained VAE encoder. During training, we will first add noise to $\epsilon(I_{HQ})$ to get the noisy latent $z_t$ and then use LAControlNet to predict the noise. You can read the following code and have a better understanding of the training process:
1. [training_step](https://github.com/XPixelGroup/DiffBIR/blob/main/ldm/models/diffusion/ddpm.py#L433) and [shared_step](https://github.com/XPixelGroup/DiffBIR/blob/main/ldm/models/diffusion/ddpm.py#L848): The class `ControlLDM` inherits these two methods from base class `DDPM` and `LatentDiffusion`, which receive batch data from dataloader and pass it to the method `get_input`. The batch data contains both hq images and the generated lq images.
2. [get_input](https://github.com/XPixelGroup/DiffBIR/blob/main/model/cldm.py#L328): The  `get_input` method of `ControlLDM` is used to preprocess the batch data. It first passes batch data to the `get_input` method of `LatentDiffusion` to encode hq images  $I_{HQ}$ to latent space (denoted as $\epsilon(I_{HQ})$ ), and then prepare condition latent $\epsilon(I_{reg})$ for LAControlNet.
3. [forward](https://github.com/XPixelGroup/DiffBIR/blob/main/ldm/models/diffusion/ddpm.py#L853): The forward method is called by `shared_step`, it first samples a timestep `t` and then call the `p_losses` method to compute loss.
4. [p_losses](https://github.com/XPixelGroup/DiffBIR/blob/main/ldm/models/diffusion/ddpm.py#L899): This method add noise to $\epsilon(I_{HQ})$ to get the noisy latent $z_t$ through the `q_sample` method and then call the method `apply_model`, which uses LAControlNet to pedict noise.

Due to the intricate inheritance relationships, the code related to LDM is difficult to understand. Please feel free to contact us if you have any further questions! 😄

**Comment by SongYxing** [comment]
Thank you so much for your patient response; it has been incredibly helpful to me.
